### PR TITLE
switch tide to clusterrole so it can create pods in other namepsaces

### PR DESCRIPTION
**Pull request by BenTheElder**
... also move to v1

/cc @fejta

### CODE REVIEW

File: prow/cluster/cluster.yaml [hunk 1]
```diff
@@ -868,10 +868,10 @@ metadata:
   namespace: default
   name: "tide"
 ---
-kind: Role
-apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
```
Comment (fejta) [inline prow/cluster/cluster.yaml]: All prow jobs live in a single namespace (the same namespace as where tide lives). So I don't understand why this is necessary?

Comment (BenTheElder) [reply prow/cluster/cluster.yaml]: the logs show tide failing to create pods in test-pods.
we're using clusterrole for our other components :man_shrugging:

Comment (BenTheElder) [reply prow/cluster/cluster.yaml]: we could create roles for each namespace instead but i'm not sure what we intend to target.

Comment (fejta) [reply prow/cluster/cluster.yaml]: Tide doesn't create pods?

Comment (BenTheElder) [reply prow/cluster/cluster.yaml]: ACK, logs read wrong, discussing in chat/

**Pull request merged**